cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/IntraBrokerDiskCapacityGoal.java

```diff
@@ -149,8 +160,12 @@ public ActionAcceptance actionAcceptance(BalancingAction action, ClusterModel cl
   protected boolean selfSatisfied(ClusterModel clusterModel, BalancingAction action) {
     Replica sourceReplica = clusterModel.broker(action.sourceBrokerId()).replica(action.topicPartition());
     Disk destinationDisk = clusterModel.broker(action.destinationBrokerId()).disk(action.destinationBrokerLogdir());
-    return sourceReplica.load().expectedUtilizationFor(RESOURCE) > 0
-           && isMovementAcceptableForCapacity(sourceReplica, destinationDisk);
+    boolean shouldMoveReplica = sourceReplica.load().expectedUtilizationFor(RESOURCE) > 0;
```

Comment:
I'd recommend to change this to:

```
    boolean shouldMoveReplica;
    if (_shouldEmptyZeroCapacityDisks) {
      // should also move replicas with 0 expected disk utilization
      shouldMoveReplica = sourceReplica.load().expectedUtilizationFor(RESOURCE) >= 0;
    } else {
     // In general CC tries to reduce number of replica move. Since empty replicas does not affect balance, CC tries to avoid moving them. 
      shouldMoveReplica = sourceReplica.load().expectedUtilizationFor(RESOURCE) > 0;
   }

```

Reply:
Good note. It looks more clear this way. Refactored in latest commit.